components/server/src/server.ts

```diff
@@ -154,14 +153,6 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
             // CORS allows subdomains to access gitpod.io)
             const verifyCSRF = (origin: string) => {
                 let allowedRequest = isAllowedWebsocketDomain(origin, this.config.hostUrl.url.hostname);
-                if (this.config.stage === "prodcopy" || this.config.stage === "staging") {
-                    // On staging and devstaging, we want to allow Theia to be able to connect to the server from this magic port
-                    // This enables debugging Theia from inside Gitpod
-                    const url = new URL(origin);
-                    if (url.hostname.startsWith("13444-")) {
-                        allowedRequest = true;
-                    }
-                }
```

Comment:
Does it matter that we are removing this debug port?

Reply:
> Does it matter that we are removing this debug port?

AFAIK we have not clients anymore. This way conceived for Theia, and is not used anymore.

Reply:
Nope, this should not be in use anymore. It's theia-specific.